ui/settings.lua

```diff
@@ -3,70 +3,112 @@ local S = minetest.get_translator("mail")
-            checkbox[0,3.6;trash_move_enable;]] .. S("Move deleted messages to trash") .. [[;]] ..
-            tostring(mail.get_setting(name, "trash_move_enable")) .. [[]
-            checkbox[0,4.0;auto_marking_read;]] .. S("Automatic marking read") .. [[;]] ..
-            tostring(mail.get_setting(name, "auto_marking_read")) .. [[]
-			label[0.31,4.7;]] .. S("Date format:") .. [[]
-            dropdown[2.7,4.6;4,0.5;date_format;]] .. date_dropdown_str .. [[;]] ..
-            tostring(date_dropdown_index) .. [[;true]
-
-            tooltip[chat_notifications;]] .. S("Receive a message in the chat when there is a new message") .. [[]
-            tooltip[onjoin_notifications;]] .. S("Receive a message at login when inbox isn't empty") .. [[]
-            tooltip[hud_notifications;]] .. S("Show an HUD notification when inbox isn't empty") .. [[]
-            tooltip[sound_notifications;]] .. S("Play a sound when there is a new message") .. [[]
-            tooltip[auto_marking_read;]] .. S("Mark a message as read when opened") .. [[]
+			tablecolumns[text]
+            table[0,0.775;3,4.5;groups;]] .. groups_str .. [[;]] .. group_index .. [[]
+
+			box[3.5,0.8;3,0.45;]] .. mail.colors.highlighted .. [[]
+			label[3.7,0.8;]] .. mail.settings_groups[group_index].label .. [[]
 
             button[0,5.65;2.5,0.5;reset;]] .. S("Reset") .. [[]
             button[7.5,5.65;2.5,0.5;save;]] .. S("Save") .. [[]
-            ]] .. mail.theme
-
+            ]]
+
+    local x = 3.5
+    local y = 1
+    -- put settings in order
+    local ordered_settings = {}
+    for setting, data in pairs(mail.settings) do
+        if data.group == mail.selected_idxs.settings_group[name] then
+            table.insert(ordered_settings, setting)
+        end
+    end
+    table.sort(ordered_settings, function(a, b) return mail.settings[a].index < mail.settings[b].index end)
+    for _, setting in pairs(ordered_settings) do
+        local data = mail.settings[setting]
+        y = y + 0.4
+        local field_default = mail.selected_idxs[setting][name]
+        if field_default == nil then field_default = mail.get_setting(name, setting) end
+        if data.type == "bool" then
+            formspec = formspec .. [[
+            checkbox[]] .. x .. "," .. y .. ";" .. setting .. ";" ..
+            data.label .. ";" .. tostring(field_default) .. [[]
+            ]]
+            if data.tooltip then
+                formspec = formspec .. [[
+                tooltip[]] .. setting .. ";" .. data.tooltip .. [[]
+                ]]
+            end
+        elseif data.type == "string" then
+            y = y + 1
+            formspec = formspec .. [[
+            field[]] .. x+0.275 .. "," .. y .. ";3,0.5;" .. setting .. ";" .. data.label .. [[;]] ..
+            field_default .. [[]
+            ]]
+            if data.tooltip then
+                formspec = formspec .. [[
+                tooltip[]] .. setting .. ";" .. data.tooltip .. [[]
+                ]]
```

Comment:
Does not have to be changed, more about just being curious if there's a reason you choose to use `[[` instead of `"`?

I mean while `[[` has its place for me `"` seems a lot cleaner:
```suggestion
                formspec = formspec .. "tooltip[" .. setting .. ";" .. data.tooltip .. "]"
```